jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ErrorHandler.java

```diff
@@ -70,7 +85,8 @@ public boolean handle(Request request, Response response, Callback callback) thr
         ServletContextHandler.ServletScopedContext context = servletContextRequest.getErrorContext();
         Integer errorStatus = (Integer)request.getAttribute(ERROR_STATUS);
         Throwable errorCause = (Throwable)request.getAttribute(ERROR_EXCEPTION);
-        if (this instanceof ErrorPageMapper mapper)
+        boolean enteredServletChannel = servletContextRequest.getServletChannel().getCallback() != null;
+        if (this instanceof ErrorPageMapper mapper && enteredServletChannel)
```

Comment:
This is a really important part of this PR, so it probably needs a comment

```suggestion
        // Error page mapping can only be supported from within the ServletChannel handling.
        // If an error that may be mapped to an error page occurs before entering ServletChannel, then the ErrorHandler#writeError(...) method should be used to delay
        // invoking sendError until the handling is within the ServletChannel
        boolean enteredServletChannel = servletContextRequest.getServletChannel().getCallback() != null;
        if (this instanceof ErrorPageMapper mapper && enteredServletChannel)
```

Actually this should probably be said in javadoc rather than a comment.  But say it somewhere.